go_covert_lib/client/src/screens/ConfigScreen.jsx

```diff
@@ -0,0 +1,307 @@
+                            },
+                          },
+                        },
+                        ...processors.slice(i + 1, processors.length + 1),
+                      ])}
+                    />
+                  );
+                case 'bool':
+                  return (
+                    <Checkbox
+                      {...propsForComponent}
+                      parentOnChange={e => setProcessors([
+                        ...processors.slice(0, i),
+                        {
+                          ...processor,
+                          Data: {
+                            ...processor.Data,
+                            [processor.Type]: {
+                              ...processor.Data[processor.Type],
+                              [key]: {
+                                ...opt,
+                                Value: e.target.checked,
+                              },
+                            },
+                          },
+                        },
+                        ...processors.slice(i + 1, processors.length + 1),
+                      ])}
+                    />
+                  );
+                case 'select':
+                  return (
+                    <Select
+                      {...propsForComponent}
+                      items={opt.Range}
+                    />
+                  );
+                default:
+                  return (<div key={key}>UNIMPLEMENTED</div>);
+              }
+            })}
+          </div>
+        ))
+      }
+      <h3 className="m-1">Channel</h3>
+      <Dropdown className="m-1">
+        <Dropdown.Toggle
+          className="cc-config__chan-select w-100"
+          variant="outline-primary"
+        >
+          {channel.value || 'Select a Channel'}
+        </Dropdown.Toggle>
+        <Dropdown.Menu className="w-100">
+          {
+            Object.keys(channelList).map(chan => (
+              <Dropdown.Item
+                as="option"
+                active={chan === channel.value}
+                onClick={(e) => {
+                  setChannel({
+                    value: e.target.value,
+                    properties: channelList[chan],
+                  });
+                  setConfig(channelList[chan]);
+                }}
+                value={chan}
+                key={chan}
+              >
+                {chan}
+              </Dropdown.Item>
+            ))
+          }
+        </Dropdown.Menu>
+      </Dropdown>
+      {Object.keys(config).map((key) => {
+        const opt = config[key];
+        let propsForComponent = {
+          key,
+          label: opt.Display.Name,
+          value: opt.Value,
+          tooltip: opt.Display.Description,
+          parentOnChange: e => setConfig({
+            ...config,
+            [key]: {
+              ...config[key],
+              Value: e.target.value,
+            },
+          }),
+        };
+        switch (opt.Type) {
```

Comment:
Does this switch statement really need to be in two places, or would it be sufficient to abstract to a single function?

Reply:
I am theoretically able to merge them together, but unfortunately there a few reasons I have not done so. Although they perform similar functions, the logic needs to be slightly different because the data is formatted slightly differently and the callbacks need to work slightly differently, based on how the backend configuration & processor list is formatted. What I am able to do is make a shared function that returns JSX (html content), but I would need to pass in a lot of data and have largely duplicated code anyway as most cases need to return something slightly different based on if it's for a processor or a configuration. So yes, it can be done, but the outcome will be very convoluted and I have therefore concluded is not worth the time of refactoring.